src/reflection.ts

```diff
@@ -688,4 +733,168 @@ export class Parser {
       return result;
     };
   }
+
+  readVectorOfUnionLambda(
+    field: reflection.Field,
+    discriminatorField: reflection.Field,
+    readDefaults: boolean,
+  ): (table: Table) => Record<string, any>[] {
+    const fieldType = field.type();
+    if (fieldType === null) {
+      throw new Error(`Malformed schema: "type" field of '${field.name()}' not populated.`);
+    }
+
+    // This will deserialize the vector of discriminator values
+    const readDiscriminators = this.readVectorOfScalarsLambda(discriminatorField);
+
+    // This will read the vector of tables from the actual field
+    // How to deserialize each table will depend on the discriminator value at the same vector
+    // index position in the discriminator values field
+    const vectorLambda = this.readVectorOfTablesLambda(field);
```

Comment:
naming nit: `readDiscriminators` and `vectorLambda` are both readers and both lambdas - we should pick one naming convention

Reply:
off topic question that came to mind: can a vector of tables contain nulls? would `readVectorOfTablesLambda` need to be updated to support that?

Reply:
At least with the types that were generated for me for tests it was not valid to push to my union vector an `undefined` value (corresponding to a NONE union enum value)